helm/hpcc/templates/_helpers.tpl

```diff
@@ -2200,3 +2200,96 @@ Fills "result" dictionary with "planeCategories" and "namedPlanes"
 {{- $_ := set .result "planeCategories" $planeCategories -}}
 {{- $_ := set .result "namedPlanes" $namedPlanes -}}
 {{- end -}}
+
+{{/*
+A template to generate a HPA Behavior.scaleUp/Down clause for HPA object
+Pass in dict behaviorscale section
+*/}}
+{{- define "hpcc.addHPABehaviorScale" }}
+{{- if .behaviorScale -}}
+ {{- if hasKey .behaviorScale "stabilizationWindowSeconds" }}
+      stabilizationWindowSeconds: {{ .behaviorScale.stabilizationWindowSeconds }}
+ {{- end }}
+ {{- if hasKey .behaviorScale "policies" }}
+      policies:
+  {{- range $policy := .behaviorScale.policies }}
+      - type: {{ $policy.type }}
+        value: {{ $policy.value }}
+        periodSeconds: {{ $policy.periodSeconds }}
+  {{- end }}
+ {{- end }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+A template to generate a HorizontalPodAutoscaler for a named workload resource (such as a Deployment or StatefulSet)
+Pass in dict with workload resource name and kind (Deployment|ReplicaSet|StatefulSet|ReplicationController), and hpa section
+*/}}
+{{- define "hpcc.addHorizontalPodAutoscaler" }}
+ {{- if .name -}}
+  {{- if .hpa -}}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ printf "%s-hpa" .name }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: {{ .kind }}
+    name: {{ printf "%s" .name }}
+  minReplicas: {{ (hasKey .hpa "minReplicas") | ternary .hpa.minReplicas 1 }}
+  maxReplicas: {{ .hpa.maxReplicas }}
+   {{- if hasKey .hpa "behavior" }}
+  behavior: 
+    {{- if hasKey .hpa.behavior "scaleDown" }}
+    scaleDown:
+      {{ include "hpcc.addHPABehaviorScale" (dict "behaviorScale" .hpa.behavior.scaleDown) }}
```

Comment:
didn't notice on last review, but to make function portable (so they can be generated at any indentation level),
we make them generate without indentation, and then apply the correct level of indentation where they are included,
e.g. in this case (with indentation removed from hpcc.addHPABehaviorScale, this line becomes:
```
    {{- if hasKey .hpa.behavior "scaleDown" }}
    scaleDown:
     {{- include "hpcc.addHPABehaviorScale" (dict "behaviorScale" .hpa.behavior.scaleDown) | indent 6 }}
    {{- end }}
```